### PR TITLE
fix(menu): enable menu items with favorites to show star icon and set icon color when favorited

### PR DESCRIPTION
**Pull request by sg00dwin**
fixes https://github.com/patternfly/patternfly/issues/7180

https://patternfly-pr-7184.surge.sh/components/menus/menu#with-favorites

### CODE REVIEW

File: src/patternfly/components/Menu/menu-item-action.hbs [hunk 1]
```diff
@@ -2,19 +2,18 @@
   {{setModifiers
     menu-list-item--IsAriaDisabled='pf-m-aria-disabled'
     menu-item-action--IsFavorite='pf-m-favorite'
```
[the remark this answers is not in the document]
Comment (sg00dwin) [reply src/patternfly/components/Menu/menu-item-action.hbs:4]: I believe you are correct, so I'll remove.

Though I did find `pf-m-favorite` [referenced in the react component](https://github.com/patternfly/patternfly-react/blob/a382cd6024f91644fb52e77090e62f743cee7eac/packages/react-core/src/components/Menu/MenuItemAction.tsx#L51), but I don't see it utilizing any styling.

**Review by srambach — APPROVED** [review]
Looks ⭐ !
Do you want to create a React followup to remove that class from the React component?

**Comment by sg00dwin** [comment]
> Looks ⭐ ! Do you want to create a React followup to remove that class from the React component?

New issue created https://github.com/patternfly/patternfly-react/issues/11176

**Review by mcoker — APPROVED** [review]
LGTM! I have a couple of small ideas as follow ups but they aren't blocking so I'll create a new issue to track those.

**Pull request merged**